fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/DataInKeySpacePathTest.java

```diff
@@ -0,0 +1,352 @@
+            byte[] keyBytes = withRemainder ? pathTuple.add("Remainder").pack() : pathTuple.pack();
+            byte[] valueBytes = Tuple.from("accessor_test").pack();
+            
+            KeyValue originalKeyValue = new KeyValue(keyBytes, valueBytes);
+            DataInKeySpacePath dataInPath = new DataInKeySpacePath(testPath, originalKeyValue, context);
+
+            // Verify resolved path future is not null
+            CompletableFuture<ResolvedKeySpacePath> resolvedFuture = dataInPath.getResolvedPath();
+            assertNotNull(resolvedFuture);
+            assertTrue(resolvedFuture.isDone() || !resolvedFuture.isCancelled());
+
+            final ResolvedKeySpacePath resolvedPath = resolvedFuture.get();
+            assertEquals(pathTuple, resolvedPath.toTuple());
+            assertArrayEquals(originalKeyValue.getValue(), dataInPath.getValue());
+            if (withRemainder) {
+                assertEquals(Tuple.from("Remainder"), resolvedPath.getRemainder());
+            } else {
+                assertNull(resolvedPath.getRemainder());
+            }
+        }
+    }
+
+    @Test
+    void withWrapper() {
+        final FDBDatabase database = dbExtension.getDatabase();
+        final EnvironmentKeySpace keySpace = EnvironmentKeySpace.setupSampleData(database);
+
+        // Test export at different levels through wrapper methods
+        try (FDBRecordContext context = database.openContext()) {
+            // Test 4: Export from specific data store level
+            final EnvironmentKeySpace.ApplicationPath appPath = keySpace.root().userid(100L).application("app1");
+            EnvironmentKeySpace.DataPath dataStore = appPath.dataStore();
+
+            final byte[] key = dataStore.toTuple(context).add("record2").add(0).pack();
+            final byte[] value = Tuple.from("data").pack();
+            final DataInKeySpacePath dataInKeySpacePath = new DataInKeySpacePath(dataStore, new KeyValue(key, value), context);
+
+            final ResolvedKeySpacePath resolvedPath = dataInKeySpacePath.getResolvedPath().join();
+            assertEquals(dataStore.toResolvedPath(context), withoutRemainder(resolvedPath));
+            assertEquals(Tuple.from("record2", 0), resolvedPath.getRemainder());
+
+            // Verify the path using assertNameAndValue  
+            // Note: We expect the path to be: [environment] -> userid -> application -> data
+            ResolvedKeySpacePath appLevel;
+            appLevel = assertNameAndValue(resolvedPath, "data", EnvironmentKeySpace.DATA_VALUE);
+            ResolvedKeySpacePath userLevel = assertNameAndDirectoryScopedValue(appLevel, "application", "app1",
+                    appPath, context);
+            ResolvedKeySpacePath envLevel = assertNameAndValue(userLevel, "userid", 100L);
+            assertNull(assertNameAndDirectoryScopedValue(envLevel, keySpace.root().getDirectoryName(),
+                    keySpace.root().getValue(), keySpace.root(), context));
+        }
+    }
+
```

Comment:
Maybe add a test that fails to resolve a key?

Reply:
I added a test that fails to resolve (has a different root)

Regarding `NULL`, are you suggesting creating a `DataInKeySpacePath` with `KeyValue(key, null)`, or a path that has a `NULL` type in it?
For the former, I don't think FDB will ever return such a `KeyValue`, and if you have one, you would never be able to insert it, because `set` does not allow a null value. Give that, it probably makes sense to have `DataInKeySpacePath` error if it gets a `null` value.

For the later, I think that should be better tested in `KeySpacePath.toResolvedPathAsync`.